tests/components/yessssms/test_notify.py

```diff
@@ -1,7 +1,144 @@
 """The tests for the notify yessssms platform."""
+# pylint: disable=redefined-outer-name
 import unittest
+from unittest.mock import MagicMock
 import requests_mock
+import pytest
+
+from homeassistant.setup import async_setup_component
 import homeassistant.components.yessssms.notify as yessssms
+from homeassistant.components.yessssms.const import CONF_PROVIDER
+
+from homeassistant.const import CONF_PASSWORD, CONF_RECIPIENT, CONF_USERNAME
+
+
+@pytest.fixture
+def config_data():
+    """Set valid config data."""
+    config = {
+        "notify": {
+            "platform": "yessssms",
+            "name": "sms",
+            CONF_USERNAME: "06641234567",
+            CONF_PASSWORD: "secretPassword",
+            CONF_RECIPIENT: "06509876543",
+            CONF_PROVIDER: "educom",
+        }
+    }
+    return config
+
+
+@pytest.fixture
+def valid_settings(hass, config_data):
+    """Initialize component with valid settings."""
+    yield async_setup_component(hass, "notify", config_data)
```

Comment:
Since we don't need to anything after yielding the function we can instead `return` the function.